sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -1258,6 +1278,10 @@ <T> long insertAll(
           }
         }
       }
+      streamingInsertsResults.failedRowsCount.getAndAdd(allErrors.size());
```

Comment:
see comment about making instance method
maybe this one could be something like:
```
@AutoValue
public abstract static StreamingInsertResults {
...
void updateRowCount(int failedRowCount, int successRowCount) {
    failedRowsCount().getAndAdd(failedRowCount);
    successfulRowsCount().set(successRowCount - failedRowsCount.get())
}
...
}
```

Reply:
Done.